consensus/construct.go

```diff
@@ -33,11 +45,22 @@ func (consensus *Consensus) construct(
 			Consensus: &msg_pb.ConsensusRequest{},
 		},
 	}
+	var (
+		consensusMsg *msg_pb.ConsensusRequest
+		aggSig       *bls.Sign
+	)
 
-	consensusMsg := message.GetConsensus()
-	consensus.populateMessageFields(consensusMsg)
-
-	var aggSig *bls.Sign
+	// if caller provided no special payload,
+	// then let the blockhash be the default off of consensus
+	if len(payloadForSignOverride) == 0 {
+		consensusMsg = consensus.populateMessageFields(
+			message.GetConsensus(), consensus.blockHash[:],
+		)
+	} else {
+		consensusMsg = consensus.populateMessageFields(
+			message.GetConsensus(), blockHash[:],
+		)
+	}
```

Comment:
should you also change the "case msg_pb.MessageType_COMMIT:" part?

Reply:
i think it currently is handled correctly